packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts

```diff
@@ -611,7 +611,7 @@ export interface TransformResources {
   /**
    * ML compute instance type for the transform job.
    */
-  readonly instanceType: ec2.InstanceType;
+  readonly instanceType: InstanceType;
```

Comment:
I know we removed the @experimental annotations after you had initially published this PR, but I believe this is now a breaking change. 

Let me check with a team member or two to see whether we would need to:
1. support the old parameter but mark it as `@deprecated`
2. make this change as it is, but call it out as a `BREAKING` change.

It is a breaking change if we ship it as it is as clients who upgrade will see breakage.

Reply:
Right, [in earlier comments](https://github.com/aws/aws-cdk/pull/14064#discussion_r638757961)I was checking whether we should make it backward compatible. Let me know if we have to extend `ec2. InstanceType` here, thanks

Reply:
Hi @shivlaks, any updates on this?

Reply:
As Shiv mentions above, we will need to create a new property (maybe `mlInstanceType`) and mark the old property as deprecated. Unfortunately we cannot extend this type with a type union due to jsii restrictions